react/introduction/setting_up_a_react_environment.md

```diff
@@ -0,0 +1,145 @@
+
+Some examples of these toolchains:
+
+* [Create React App](https://create-react-app.dev/)
+* [Vite's React Config](https://vitejs.dev/)
+* [Gatsby](https://www.gatsbyjs.com/)
+* [NextJS](https://nextjs.org/)
+
+Why do we need these toolchains? Can't we just make our own as we see fit?
+
+Yes, but it's _hard_. React is a complex beast and there are many moving parts. Before you can start writing any sort of code that provides functionality, you would need to configure _at least_ the following:
+
+* Package Management (NPM, Yarn)
+* Module bundling (Webpack, Parcel)
+* Compilation (Babel)
+* React itself
+
+All of this, and sometimes _much more_ is required to get a React project and development environment up and running.
+
+### Simplifying The Process
+
+Now that you understand what is involved with starting a React project from scratch, you can breathe a sigh of relief as you learn that we can get started with a _single terminal command_.
+
+This terminal command is a feature of the beginner-friendly [Create React App](https://create-react-app.dev/), a tool developed by the folks over at Meta which we will be using from here on out. It requires minimal configuration and provides extremely useful tools straight out of the box, allowing us to get straight to the learning. Let's get started!
+
+### Creating A React App
+
+Fire up a terminal instance, `cd` over to the folder containing your projects and enter the following command:
+
+~~~
+$ npx create-react-app my-first-react-app
+~~~
+
+Note that we use `npx` here (not `npm`), this is not a typo, `npx` is a package runner that allows the execution of packages without them being installed locally on your machine. [Check here](https://www.geeksforgeeks.org/what-are-the-differences-between-npm-and-npx/) for more information.
+
+Once this process has completed, we should confirm that all has gone well:
+
+~~~
+$ cd my-first-react-app
+$ npm start
+~~~
+
+Provided that everything has gone according to plan, you should be greeted with a new browser tab containing a lovely spinning React logo:
+
+![Create React App Starting Page](./setting_up_a_react_environment/imgs/cra_start.png)
+
+Congratulations! You've created your first React app.
+
+### Delving Deeper
+
+Let's take a closer look at our new project. Inside you will find some folders, as well as `package.json`, `package-lock.json`, `.gitignore`, and `README.md` files. The `README.md` contains some useful information that you should take the time to skim through now.
+
+The `public` folder is where all of the static assets related to your app will go. This could include images, icons, information files for the browser, and perhaps most importantly—the `index.html` file.
+
+Inside of the `src` folder is where you will find the code that runs your app. Let's open the `index.js` file and see if we can understand what's going on:
```

Comment:
Some information about the `index.js` file. I think we should include this info; this lets us justify why we're looking at the code inside the file. 

```suggestion
Inside of the `src` folder is where you will find the code that runs your app. `index.js` is the entry point of the application. Let's open the `index.js` file and see if we can understand what's going on:
```

Reply:
Added this change with very slightly different wording, let me know if you'd like me to use the exact language here.